edgelet/edgelet-core/src/secret.rs

```diff
@@ -0,0 +1,45 @@
+use std::fmt;
+
+use failure::Fail;
+use futures::Future;
+
+pub trait SecretManager {
+    type Error: Fail;
+    type SetFuture: Future<Item = (), Error = Self::Error> + Send;
```

Comment:
except GetFuture, all other Future types are the same, why not reused?

Reply:
I mirrored the style used by the other modules in `edgelet-core`: [example](https://github.com/Azure/iotedge/blob/aae4917d9adf726b2450a988fc25a04df2e84bd9/edgelet/edgelet-core/src/module.rs#L497-L500).